### DISCUSSION TITLE
catchboundary not working, only error boundary works

### DISCUSSION THREAD

**Discussion by shop-fluencer** [remix-run/remix#5055] [Q&A]
### What version of Remix are you using?

1.10

### Steps to Reproduce

```
import { ActionArgs, redirect, json } from '@remix-run/node';
import { useCatch, useLoaderData } from '@remix-run/react';
import NewNote, { links as newNoteLinks } from '~/components/NewNote';
import NoteList from '~/components/NoteList';
import { getStoredNotes, storeNotes } from '../data/note';

export default function Notes() {
  const notes = useLoaderData();
  return (
    <main>
      <h1>My Notes</h1>
      <NewNote />
      <NoteList notes={notes} />
    </main>
  );
};


export async function loader() {
 const notes = await getStoredNotes();

 if(!notes) {
  throw json({message: 'Could not find any notes.'}, {
    status: 404,
    statusText: 'Not found.'
  });
}
 
 return notes;
};

export function CatchBoundary() {
  const caughtResponse = useCatch();


  return (
    <p>{caughtResponse.statusText}</p>
  )
}

export function ErrorBoundary({error}) {
  return (
    <main>
      <h1>ERROR</h1>
    </main>
  )
}

export async function action({ request }: ActionArgs) {
 const formData = await request.formData();
 const noteData = Object.fromEntries(formData);
 // Add validation
 if(noteData.title.trim().length < 5) {
  return {
    message: 'Invalid title'
  }
 }
 const existingNotes = await getStoredNotes();
 noteData.id = new Date().toISOString();
 const updatedNotes = existingNotes.concat(noteData);
 await storeNotes(updatedNotes);

 return redirect('/notes');
}

export function links() {
  return [...newNoteLinks()];
}
```

### Expected Behavior

Why my catch boundary not working ? Only error boundary works

### Actual Behavior

..

€: I renamed an file so the file where I update note does not exists, to test if catch works

**Comment by kiliman** [comment]
Hmm.. I'm not seeing this.
```ts
export async function loader() {
  throw json(
    { message: "Could not find any notes." },
    {
      status: 404,
      statusText: "Not found.",
    }
  );
}
export function CatchBoundary() {
  const caughtResponse = useCatch();
  return (
    <>
      <p>{caughtResponse.statusText}</p>
      <pre>{JSON.stringify(caughtResponse.data, null, 2)}</pre>
    </>
  );
}
```
<img width="358" alt="image" src="https://user-images.githubusercontent.com/47168/211666346-7738889c-2e1f-431a-b36f-8a36ae8ccff1.png">

And I'm on v1.10
<img width="633" alt="image" src="https://user-images.githubusercontent.com/47168/211666504-e9dba393-650b-48b6-b374-08c03d17c129.png">

**Comment by kiliman** [comment]
Can you create a code sandbox or a stack blitz?

**Comment by shop-fluencer** [comment]
> Can you create a code sandbox or a stack blitz?

create a node.json file on / folder and make a action or loader and rename this node.json to node.jso so that he cant find the file

**Comment by kiliman** [comment]
I'm not sure what you mean about a `node.json` file.

**Comment by rperon** [comment]
@kiliman 

I'm experiencing the same issue when updating my app to `remix 1.10`, my CatchBoundary no longer works.

I tried to reproduce it on a fresh remix app, but it's behaving correctly... But I'm able to reproduce it with a blue stack. (my app is based on the blue stack.)

<img width="1279" alt="Screenshot 2023-01-12 at 19 21 51" src="https://user-images.githubusercontent.com/692098/212147768-d0a82c71-f318-407e-962c-e4d28b58b1cd.png">

Here is the link to the [CodeSandbox](https://codesandbox.io/p/github/rperon/bluecatch/draft/mutable-monad?create=true&file=%2Fapp%2Froot.tsx&selection=%5B%7B%22endColumn%22%3A17%2C%22endLineNumber%22%3A29%2C%22startColumn%22%3A17%2C%22startLineNumber%22%3A29%7D%5D&workspace=%257B%2522activeFileId%2522%253A%2522clcteumwu00138rg75nz9bvx4%2522%252C%2522openFiles%2522%253A%255B%2522%252FREADME.md%2522%252C%2522%252Fapp%252Fdb.server.ts%2522%252C%2522%252Fapp%252Froutes%252Findex.tsx%2522%252C%2522%252Fapp%252Froot.tsx%2522%255D%252C%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522gitSidebarPanel%2522%253A%2522COMMIT%2522%252C%2522spaces%2522%253A%257B%2522clcteuoht001d3b5xa7ajwuyo%2522%253A%257B%2522key%2522%253A%2522clcteuoht001d3b5xa7ajwuyo%2522%252C%2522name%2522%253A%2522Default%2522%252C%2522devtools%2522%253A%255B%257B%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clctewk0r00268rg7cywlh03o%2522%252C%2522key%2522%253A%2522clctewjxc00mn3b5x63rsil72%2522%252C%2522isMinimized%2522%253Afalse%257D%255D%257D%257D%252C%2522currentSpace%2522%253A%2522clcteuoht001d3b5xa7ajwuyo%2522%252C%2522spacesOrder%2522%253A%255B%2522clcteuoht001d3b5xa7ajwuyo%2522%255D%257D)

**Reply by adelrodriguez** [reply]
I was running into this issue as well, it turned out I had updated several packages to 1.10 but forgot to update the `@remix-run/dev` package. After updating that, the `CatchBoundary` worked.

**Reply by rperon** [reply]
Thanks for the hint, I changed in my package.json from ` ^1.10.0` to `1.10.0` for every remix package in my package.json.

Unfortunately, I still experiencing the same error...

**Reply by rperon** [reply]
So I figured out which element is messing with CatchBoundary and it appears that it is my `server.ts` file !

By switching from a `server.ts` to a `server.js`, `catchBoundary` were working as expected...

In my package.json I'm building my server this way :
```
"build:server": "esbuild --platform=node --format=cjs ./server.ts --outdir=build --bundle",
```

and I'm using it like this :

```
"dev:server": "cross-env NODE_ENV=development node --inspect --require ./node_modules/dotenv/config --require ./mocks ./build/server.js",
```

It's very similar with what you can find in the blue stack... I guess something is wrong with the esbuild but I don't know why.

**Reply by robmonie** [reply]
If anyone else is following this, it appears to be fixed in `1.12.0` - see this discussion - https://github.com/remix-run/blues-stack/issues/145
